lms/migrations/versions/a7e9dc7c4c0f_backfill_lms_segment_groups.py

```diff
@@ -0,0 +1,63 @@
+"""Backfill lms_segment (groups)."""
+
+import sqlalchemy as sa
+from alembic import op
+
+revision = "a7e9dc7c4c0f"
+down_revision = "2f7913c24b69"
+
+
+def upgrade() -> None:
+    conn = op.get_bind()
+    conn.execute(
+        sa.text("""
+        WITH backfill as (
+            -- Deduplicate "grouping" sections on authority_provided_id
+            SELECT DISTINCT ON (grouping.authority_provided_id)
+                "grouping".created,
+                "grouping".updated,
+                grouping.authority_provided_id,
+                grouping.lms_name,
+                grouping.lms_id,
+                "lms_course".id as lms_course_id,
+                grouping.type,
+                lms_group_set.id as lms_group_set_id
+            FROM "grouping"
+            -- Join grouping on itself to get the parent (course) data
+            JOIN "grouping" parent on parent.id = grouping.parent_id
+            -- Get the right LMSCourse row based on the parent
+            JOIN lms_course on lms_course.h_authority_provided_id = parent.authority_provided_id
+            LEFT OUTER JOIN lms_group_set ON lms_group_set.lms_course_id = lms_course.id and lms_group_set.lms_id = grouping.extra->>'group_set_id'::text
```

Comment:
Main difference with the sections backfill, we join with group_sets to get the right ID.

We could have done it one go but I find this a bit easier to follow.